modules/tensor_mechanics/include/materials/ComputeCappedWeakPlaneStress.h

```diff
@@ -0,0 +1,357 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef COMPUTECAPPEDWEAKPLANESTRESS_H
+#define COMPUTECAPPEDWEAKPLANESTRESS_H
+
+#include "ComputeStressBase.h"
+#include "TensorMechanicsHardeningModel.h"
+
+class ComputeCappedWeakPlaneStress;
+
+template<>
+InputParameters validParams<ComputeCappedWeakPlaneStress>();
+
+/**
+ * ComputeCappedWeakPlaneStress performs the return-map
+ * algorithm and associated stress updates for plastic
+ * models that describe capped weak-plane plasticity
+ *
+ * It assumes various things about the elasticity tensor.
+ * E(i,i,j,k) = 0 except if k=j
+ * E(0,0,i,j) = E(1,1,i,j)
+ */
+class ComputeCappedWeakPlaneStress :
+  public ComputeStressBase
+{
+public:
+  ComputeCappedWeakPlaneStress(const InputParameters & parameters);
+
+protected:
+  virtual void computeQpStress() override;
+  virtual void initQpStatefulProperties() override;
+
+  /// Hardening model for cohesion
+  const TensorMechanicsHardeningModel & _cohesion;
+
+  /// Hardening model for tan(phi)
+  const TensorMechanicsHardeningModel & _tan_phi;
+
+  /// Hardening model for tan(psi)
+  const TensorMechanicsHardeningModel & _tan_psi;
+
+  /// Hardening model for tensile strength
+  const TensorMechanicsHardeningModel & _tstrength;
+
+  /// Hardening model for compressive strength
+  const TensorMechanicsHardeningModel & _cstrength;
+
+  /// The cone vertex is smoothed by this amount
+  const Real _small_smoother2;
+
+  /// Maximum number of Newton-Raphson iterations allowed in the return-map process
+  const unsigned _max_nr_its;
+
+  /// Whether to perform finite-strain rotations
+  const bool _perform_finite_strain_rotations;
+
+  /// Smoothing tolerance: edges of the yield surface get smoothed by this amount
+  const Real _smoothing_tol;
+
+  /// The yield-function tolerance
+  const Real _f_tol;
+
+  /// Square of the yield-function tolerance
+  const Real _f_tol2;
+
+  /// The type of tangent operator to return.  tangent operator = d(stress_rate)/d(strain_rate).
+  enum TangentOperatorEnum {
```

Comment:
Minor: You might start switching to the new C++11 enum syntax. It's basically the same except for the keyword class. This creates a namespace when you use the enumeration items which I think looks a lot cleaner.